src/main/kotlin/ch/unibas/dmi/dbis/cottontail/database/serializers/ComplexValueSerializer.kt

```diff
@@ -0,0 +1,23 @@
+package ch.unibas.dmi.dbis.cottontail.database.serializers
+
+import ch.unibas.dmi.dbis.cottontail.model.values.ComplexValue
+import ch.unibas.dmi.dbis.cottontail.model.values.complex.Complex
+import org.mapdb.DataInput2
+import org.mapdb.DataOutput2
+import org.mapdb.Serializer
+
+object ComplexValueSerializer : Serializer<ComplexValue> {
+    override fun deserialize(input: DataInput2, available: Int): ComplexValue = ComplexValue(input.readComplex())
+    override fun serialize(out: DataOutput2, value: ComplexValue) {
+        out.writeComplex(value.value)
+    }
+}
+
+fun DataInput2.readComplex(): Complex {
+    // TODO readComplex
+    return Complex(0.0f, 0.0f)
+}
+
+fun DataOutput2.writeComplex(complex: Complex) {
```

Comment:
You can implement this as an extension function but it is not really necessary. What you need to do here, is simply serialize the individual components (i.e. the real and imaginary part). That is, with your current implementation:

`
 out.writeFloat(complex.real)
 out.writeFloat(complex.imaginary)
`

However, first review my comments on your Value<Complex> implementation, as this may impact the implementation.